code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -3893,6 +3893,91 @@
 	new /mob/living/simple_animal/hostile/ginger/gingerboneman(L)
 	qdel(holder.my_atom)
 
+/datum/chemical_reaction/aminoblatella
+	name = "Aminoblatella"
+	id = AMINOBLATELLA
+	result = AMINOBLATELLA
+	required_reagents = list(AMINOMICIN = 1, ROACHSHELL = 5)
+	result_amount = 1
+
+/datum/chemical_reaction/synthroach
+	name = "Synthroach"
+	id = "synthroach"
+	result = null
+	required_reagents = list(NUTRIMENT = 1, AMINOBLATELLA = 1)
+	result_amount = 1
+	
+/datum/chemical_reaction/synthroach/on_reaction(var/datum/reagents/holder, var/created_volume)
```

Comment:
the fact that 3 chems now paste this same code block is horrifying

use inheritance! make them all children of one synth animal reaction that calls a new proc called something like `spawn_animals()`

Then each one of these chemicals just has to have a couple of lines in said proc spawning its respective thing

Reply:
That's out of scope for the PR, I just want to add a use for roach chitin, not rework how the way the aminomicin chems work.

Reply:
Don't paste code, it's not out of scope, this doesn't require you to write anything new, it just requires you to implement some basic OOP

We can't keep kicking this can down the road.

Reply:
We both know that's way outside my field, but sure. How would I do it?

Reply:
Would making "aminoreaction" as a parent, moving all but the spawning itself to that parent, then making aminoreaction/child work?

Reply:
I'm testing something right now. It kinda works.

Reply:
You would do it just like I said

Make them all children of the same chemical reaction so you can use the same on_reaction, but instead of spawning the animal at the end of the code block, call another proc (name it whatever you want), then in each of the children you use that new block to spawn the animals

Reply:
I was testing a method on localhost and it worked, added it to the PR.
Check again.